### PR TITLE
VCC2018 recipe

### PR DESCRIPTION
**Pull request by unilight**
Recipe for training VQVAE on VCC2018.

1. I found that the config files for VCC2020 are gone (due to rebase?) so I added them.

2. Also, I found that `*.yml` files are ignored.
https://github.com/k2kobayashi/crank/blob/master/egs/vaevc/.gitignore#L5
Is there a specific reason?

3. About your comment on the recipe design, 
> We supposed to share run.sh over all recipes (i.e., use symbolic link) because of reducing maintenance const for further recipes. Are there any difficulty to do that? If so, it will require to modify template/run.sh.

Currently, I am okay with sharing recipes. Creating new `run.sh` files is just a habit from ESPnet. On the other hand, creating a new `run.sh` for every dataset can make the design more flexible. Flexibility might become an issue as we develop more models and train on more datasets. But anyway this is all about personal choice.

### CODE REVIEW

**Comment by k2kobayashi** [comment]
For the repository policy, It should keep our templates to share all recipe.
By retaining such structure, when we add new features such as neural vocoder, it is straightforward to implement it because we just modify and test just once.
So, when creating a new recipe, we just modify ```egs/vaevc/<recipe>/{conf,local,data}``` .

Of course, this constraint makes flexibility degradation. But, when it happens, we can create new method (i.e., move on egs/<new_feature> directory). 
In terms of contribution for recipe creation, it'll describe their usernames on REAMD.md.

**Comment by unilight** [comment]
I changed to use `template/run.sh`. Also, I changed so that `download.sh` takes `downloaddir` as input instead of `wavdir` so we can save other files, ex. transcriptions. This is for, ex. ASR evaluation.
It should be okay to merge now.

File: egs/vaevc/template/run.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ n_jobs=10 # number of parallel jobs
 n_gpus=1  # number of gpus
 
 # directory settings
-wavdir=downloads/wav # directory to save downloaded wav files
+downloaddir=        # directory to save downloaded wav files
```
Comment (k2kobayashi) [inline egs/vaevc/template/run.sh]: When $downloaddir is not defined, $wavdir is placed in ```/wav``` not ```./wav``` isn't it?

File: egs/vaevc/vcc2018v1/conf/default.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
+# feature setting
```
Comment (k2kobayashi) [inline egs/vaevc/vcc2018v1/conf/default.yml]: Is it ok to make default.yml as symbolic link from ```template/conf```? Maybe you need to overwrite it by ```vcc2020v1/conf/default.yml```.

File: egs/vaevc/vcc2018v1/conf/slurm.conf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+command sbatch --export=PATH  --ntasks-per-node=1
```
Comment (k2kobayashi) [inline egs/vaevc/vcc2018v1/conf/slurm.conf:1]: Is it ok to as symbolic link? I did not try.

**Review by k2kobayashi — COMMENTED** [review]
Thanks! I have a few comments related to create a new recipe. Could you include such as modification to this PR?

File: egs/vaevc/template/run.sh (hunk 1 above)
Comment (unilight) [reply egs/vaevc/template/run.sh]: I see. I will specify a default value for `downloaddir`.

File: egs/vaevc/vcc2018v1/conf/default.yml (hunk 2 above)
Comment (unilight) [reply egs/vaevc/vcc2018v1/conf/default.yml]: Sorry, I don't really understand... the current design is to put separate configs for each dataset, right?

File: egs/vaevc/vcc2018v1/conf/slurm.conf (hunk 3 above)
Comment (unilight) [reply egs/vaevc/vcc2018v1/conf/slurm.conf:1]: Same as above.

File: egs/vaevc/vcc2018v1/conf/default.yml (hunk 2 above)
Comment (k2kobayashi) [reply egs/vaevc/vcc2018v1/conf/default.yml]: Sorry, it requires to change feature extraction settings in each recipe.
I thought the word of default should be default for vaevc not each recipe, but it's my misunderstandings.

Comment (unilight) [reply egs/vaevc/vcc2018v1/conf/default.yml]: I see. Or shall we use more meaningful names? For example, if we want to compare different network types or features, we can name it to something like `mlfb.yml`, `dilatedcnn.yml`?

Comment (k2kobayashi) [reply egs/vaevc/vcc2018v1/conf/default.yml]: That's good idea. ```${feat_type}_${trainer_type}.yml``` like ```{mlfb}_{vqvae}.yml``` is fine.

File: egs/vaevc/template/run.sh [hunk 4]
```diff
@@ -45,6 +45,7 @@ eval_speakers=""  # evaluation speaker
 set -eu # stop when error occured and undefined vars are used
 
 mkdir -p "${expdir}"
+wavdir=${downloadir}/wav
```
Comment (k2kobayashi) [inline egs/vaevc/template/run.sh]: downloadir
->
downloaddir

File: egs/vaevc/template/run.sh [hunk 5]
```diff
@@ -53,7 +54,7 @@ logdir=${datadir}/log; mkdir -p ${logdir}
 if [ "${stage}" -le 0 ] && [ "${stop_stage}" -ge 0 ]; then
     echo "stage 0: download dataset and generate scp"
     ${train_cmd} "${logdir}/download.log" \
-        local/download.sh --wavdir "${wavdir}"
+        local/download.sh "${downloaddir}"
```
Comment (k2kobayashi) [inline egs/vaevc/template/run.sh]:
```
local/download.sh "${downloaddir}"
->
local/download.sh \
  --downloaddir "${downloaddir}"
```
For kaldi-style arguments of shell scripts, it uses parse_options.sh which enables to automatically deploy arguments to named variable. You also need to modify ```local/download.sh```

File: egs/vaevc/vcc2018v1/local/download.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
+#! /bin/bash
+#
+# download.sh
+# Copyright (C) 2020 Wen-Chin HUANG
+#
+# Distributed under terms of the MIT license.
+#
+
+db_root=$1
```
Comment (k2kobayashi) [inline egs/vaevc/vcc2018v1/local/download.sh]:
```
db_root -> downloaddir
```
need to modify all $db_root variables.

**Review by k2kobayashi — COMMENTED** [review]
I ran and checked download scripts. followings are some comments.

File: egs/vaevc/vcc2018v1/local/download.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    for spk in vcc2018_evaluation/VCC2*; do
+        cp ${spk}/*.wav ${wavdir}/$(basename ${spk})
+    done
+    for spk in vcc2018_reference/VCC2*; do
+        cp ${spk}/*.wav ${wavdir}/$(basename ${spk})
+    done
+    
+    # rename the folders
+    cd ${wavdir}
+    rename "s/VCC2//g" *
```
Comment (k2kobayashi) [inline egs/vaevc/vcc2018v1/local/download.sh]: I do not have ```rename``` command. Is it possible to change it by substitute?

Comment (unilight) [reply egs/vaevc/vcc2018v1/local/download.sh]: Can I know what platform you are using? I didn't know that `rename` is platform-specific...

Comment (k2kobayashi) [reply egs/vaevc/vcc2018v1/local/download.sh]: Ubuntu 18.04.4 LTS as well as Mac OSX.

Comment (unilight) [reply egs/vaevc/vcc2018v1/local/download.sh]: Okay. I changed to use `mv`.

**Comment by k2kobayashi** [comment]
Thanks and good job @unilight. I merged.

**Pull request merged**